pkg/controllers/nodeclass/nodeclass_test.go

```diff
@@ -790,4 +793,138 @@ var _ = Describe("NodeClassController", func() {
 			Expect(nodeClass.Annotations[v1beta1.AnnotationNodeClassHash]).To(Equal(expectedHash))
 		})
 	})
+	Context("NodeClass Termination", func() {
+		var profileName string
+		BeforeEach(func() {
+			nodeClass.Spec.Role = "test-role"
+			ExpectApplied(ctx, env.Client, nodeClass)
+			profileName = instanceprofile.GetProfileName(ctx, nodeClass)
+		})
+		It("should succeed to delete the instance profile with no associated instances", func() {
+			awsEnv.IAMAPI.InstanceProfiles = map[string]*iam.InstanceProfile{
+				profileName: {
+					InstanceProfileName: aws.String(profileName),
+					Roles: []*iam.Role{
+						{
+							RoleId:   aws.String(fake.RoleID()),
+							RoleName: aws.String(nodeClass.Spec.Role),
+						},
+					},
+				},
+			}
+			ExpectReconcileSucceeded(ctx, nodeClassController, client.ObjectKeyFromObject(nodeClass))
+			Expect(awsEnv.IAMAPI.InstanceProfiles).To(HaveLen(1))
+
+			Expect(env.Client.Delete(ctx, nodeClass)).To(Succeed())
+			ExpectReconcileSucceeded(ctx, nodeClassController, client.ObjectKeyFromObject(nodeClass))
+			Expect(awsEnv.IAMAPI.InstanceProfiles).To(HaveLen(0))
+			ExpectNotFound(ctx, env.Client, nodeClass)
+		})
+		It("should succeed to delete the instance profile when no roles exist with no associated instances", func() {
+			awsEnv.IAMAPI.InstanceProfiles = map[string]*iam.InstanceProfile{
+				profileName: {
+					InstanceProfileName: aws.String(profileName),
+				},
+			}
+			ExpectReconcileSucceeded(ctx, nodeClassController, client.ObjectKeyFromObject(nodeClass))
+			Expect(awsEnv.IAMAPI.InstanceProfiles).To(HaveLen(1))
+
+			Expect(env.Client.Delete(ctx, nodeClass)).To(Succeed())
+			ExpectReconcileSucceeded(ctx, nodeClassController, client.ObjectKeyFromObject(nodeClass))
+			Expect(awsEnv.IAMAPI.InstanceProfiles).To(HaveLen(0))
+			ExpectNotFound(ctx, env.Client, nodeClass)
+		})
+		It("should succeed to delete the instance profile when it doesn't exist", func() {
```

Comment:
nit: think this semantically makes more sense as a sentence :) 
```suggestion
		It("should succeed to delete the node class when the instance profile doesn't exist", func() {
```